kubernetes/template/DeploymentTemplates/script.sh

```diff
@@ -208,6 +219,70 @@ apt_get_install()
     wait_for_apt_locks
 }
 
+generate_api_model()
+{
+    touch $DEFINITION_TEMPLATE_NAME
+    cat > $DEFINITION_TEMPLATE_NAME <<EOL
+{
+    "apiVersion": "vlabs",
+    "location": "",
+    "properties": {
+        "orchestratorProfile": {
+            "orchestratorType": "Kubernetes",
+            "orchestratorRelease": "",
+            "orchestratorVersion":"",
+            "kubernetesConfig": {
+                "useInstanceMetadata": false,
+                "networkPlugin": "",
+                "containerRuntime": "",
+                "kubeletConfig": {
```

Comment:
Any reason for not including `cloudProvider` settings?
https://github.com/Azure/aks-engine/blob/master/examples/azure-stack/kubernetes-azurestack.json